checkers/sqlQuery_checker.go

```diff
@@ -0,0 +1,152 @@
+package checkers
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/go-lintpack/lintpack"
+	"github.com/go-lintpack/lintpack/astwalk"
+	"github.com/go-toolsmith/astcast"
+)
+
+func init() {
+	var info lintpack.CheckerInfo
+	info.Name = "sqlQuery"
+	info.Tags = []string{"diagnostic", "experimental"}
+	info.Summary = "Detects issue in Query() and Exec() calls"
+	info.Before = `_, err := db.Query("UPDATE ...")`
+	info.After = `_, err := db.Exec("UPDATE ...")`
+
+	collection.AddChecker(&info, func(ctx *lintpack.CheckerContext) lintpack.FileWalker {
+		return astwalk.WalkerForStmt(&sqlQueryChecker{ctx: ctx})
+	})
+}
+
+type sqlQueryChecker struct {
+	astwalk.WalkHandler
+	ctx *lintpack.CheckerContext
+}
+
+func (c *sqlQueryChecker) VisitStmt(stmt ast.Stmt) {
+	assign := astcast.ToAssignStmt(stmt)
+	if len(assign.Lhs) != 2 { // Query() has 2 return values.
+		return
+	}
+	if len(assign.Rhs) != 1 {
+		return
+	}
+
+	call := astcast.ToCallExpr(assign.Rhs[0])
+	funcExpr := astcast.ToSelectorExpr(call.Fun)
+	if !c.funcIsQuery(funcExpr) {
+		return
+	}
+
+	// If Query() is called, but first return value is ignored,
+	// there is no way to close/read the returned rows.
+	// This can cause a connection leak.
+	if id, ok := assign.Lhs[0].(*ast.Ident); ok && id.Name != "_" {
+		return
+	}
+
+	if c.typeHasExecMethod(c.ctx.TypesInfo.TypeOf(funcExpr.X)) {
+		c.warnAndSuggestExec(funcExpr)
+	} else {
+		c.warnRowsIgnored(funcExpr)
+	}
+}
+
+func (c *sqlQueryChecker) typeIsRowsLike(typ types.Type) bool {
+	switch typ := typ.(type) {
+	case *types.Pointer:
+		return c.typeIsRowsLike(typ.Elem())
+	case *types.Named:
+		switch typ.Obj().Name() {
+		case "Rows", "Row":
+			return true
+		}
+	}
+	return false
+}
+
+func (c *sqlQueryChecker) funcIsQuery(funcExpr *ast.SelectorExpr) bool {
+	switch funcExpr.Sel.Name {
+	case "Query":
+		// Stdlib and friends.
+	case "Queryx", "QueryRowx":
```

Comment:
QueryRowx is redundant, because a returned value with a type Row hasn't Close() method

Also I suggest to add this methods:
```
QueryContext
QueryxContext
```

Reply:
Removed QueryRowx.
Added QueryContext and QueryxContext.